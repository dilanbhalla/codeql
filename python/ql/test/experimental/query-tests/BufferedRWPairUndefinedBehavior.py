import io

def buffer_rw_pair_test():

    reader = io.BufferedReader(io.RawIOBase)
    # BAD, uses reader for both parameters
    io.BufferedRWPair(reader, reader)

    writer = io.BufferedWriter(io.RawIOBase)
    # GOOD, uses different reader/writer
    io.BufferedRWPair(reader, writer)