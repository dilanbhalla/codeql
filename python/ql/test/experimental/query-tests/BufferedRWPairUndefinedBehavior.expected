| BufferedRWPairUndefinedBehavior.py:7:13:7:45 | ControlFlowNode for Attribute() | BufferedRWPair does not attempt to synchronize accesses to its underlying raw streams |
